File: backend/app/api/v1/triage.py
from fastapi import APIRouter, HTTPException, Depends
from sqlalchemy.orm import Session
from app.ai.model import run_triage_remote
from app.schemas.triage import TriageRequest, TriageResponse
from app.api.deps import get_db, get_current_user
from app.db import models

router = APIRouter()

@router.post("/triage", response_model=TriageResponse)
async def triage(
    req: TriageRequest,
    db: Session = Depends(get_db),
    current_user: models.User = Depends(get_current_user),
):
    try:
        result = await run_triage_remote(
            symptoms   = req.symptoms,
            free_text  = req.free_text or "",
            age        = req.age,
            sex        = req.sex,
            conditions = req.conditions or [],
        )
    except Exception as e:
        raise HTTPException(status_code=502, detail=f"MedGemma error: {e}")

    # Save to DB
    assessment = models.Assessment(
        user_id    = current_user.id,
        symptoms   = req.symptoms,
        free_text  = req.free_text,
        age        = req.age,
        sex        = req.sex,
        conditions = req.conditions,
        priority   = result["priority"],
        tagline    = result.get("tagline"),
        reason     = result.get("reason"),
        next_steps = result.get("next_steps", []),
    )
    db.add(assessment)
    db.commit()
    db.refresh(assessment)

    return TriageResponse(
        priority      = result["priority"],
        tagline       = result.get("tagline"),
        reason        = result.get("reason"),
        next_steps    = result.get("next_steps", []),
        assessment_id = assessment.id,
    )

File: frontend/acuvia_app/lib/data/models/triage.dart
class TriageRequest {
  final int age;
  final String sex;
  final List<String> symptoms;
  final List<String> conditions;
  final String freeText;

  const TriageRequest({
    required this.age,
    required this.sex,
    required this.symptoms,
    required this.conditions,
    required this.freeText,
  });

  factory TriageRequest.fromJson(Map<String, dynamic> json) {
    return TriageRequest(
      age: json['age'] as int,
      sex: json['sex'] as String,
      symptoms: List<String>.from(json['symptoms'] ?? []),
      conditions: List<String>.from(json['conditions'] ?? []),
      freeText: json['free_text'] as String,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'age': age,
      'sex': sex,
      'symptoms': symptoms,
      'conditions': conditions,
      'free_text': freeText,
    };
  }
}

class TriageResult {
  final int assessmentId;
  final String priority;
  final String? tagline;
  final String? reason;
  final List<String> nextSteps;

  const TriageResult({
    required this.assessmentId,
    required this.priority,
    this.tagline,
    this.reason,
    required this.nextSteps,
  });

  factory TriageResult.fromJson(Map<String, dynamic> json) {
    return TriageResult(
      assessmentId: json['assessment_id'] as int,
      priority: json['priority'] as String,
      tagline: json['tagline'] as String?,
      reason: json['reason'] as String?,
      nextSteps: List<String>.from(json['next_steps'] ?? []),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'assessment_id': assessmentId,
      'priority': priority,
      'tagline': tagline,
      'reason': reason,
      'next_steps': nextSteps,
    };
  }
}